src/tests/unittests/utils/test_serialization.py

```diff
@@ -21,11 +21,15 @@
 import unittest
 import numpy as np
 
-from qilib.utils.serialization import serialize, unserialize
+from qilib.utils.serialization import serializer, serialize, unserialize, Encoder, Decoder
 
 
-class TestSerialization(unittest.TestCase):
+class CustomType:
+    def __init__(self, x, y):
+        self.x, self.y = x, y
+
 
+class TestSerialization(unittest.TestCase):
```

Comment:
No checks on non-str keys in dicts behaviour.

Reply:
Ticket doesn't say anything about this. I think DEM-1071 deals with this.

Reply:
DEM-1071 is only about MongoDB. The new serializer does have some very specific behaviour for non-str keys:
```
> a=qilib.utils.PythonJsonStructure({'a': 1, 1: 'a'})
> j=qilib.utils.serialization.serialize(a)
> print(j)
{"a": 1, "1": "a"}
```
So `unserialize(serialize({1:'a'}))!={1: 'a'}`. This behaviour is not wanted, but if we accept it then at least it should be tested (and documented)

Something like `qilib.utils.serialization.unserialize({np.array([1,2]): 'a'})` will throw a warning

Reply:
There are no checks indeed, as there weren't in the previous version. I think we should accept it for now and make a feature request if we actually need this feature. I will add a test for it and document it.

Reply:
Also, `{np.array([1,2]): 'a'}` would never work with Python, as a numpy array is a non-hashable type.